subsys/suit/recovery_button/src/suit_recovery_button.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <nrfx_gpiote.h>
+#include <sdfw/sdfw_services/suit_service.h>
+
+#define RECOVERY_BUTTON_NODE DT_CHOSEN(recovery_button)
+#define RECOVERY_BUTTON_PIN DT_GPIO_PIN(RECOVERY_BUTTON_NODE, gpios)
+#define RECOVERY_BUTTON_PORT_NUM DT_PROP(DT_GPIO_CTLR(RECOVERY_BUTTON_NODE, gpios), port)
+#define RECOVERY_BUTTON_FLAGS DT_GPIO_FLAGS(RECOVERY_BUTTON_NODE, gpios)
+
+#define RECOVERY_BUTTON_ABS_PIN NRF_GPIO_PIN_MAP(RECOVERY_BUTTON_PORT_NUM, RECOVERY_BUTTON_PIN)
+#define RECOVERY_BUTTON_PULL (RECOVERY_BUTTON_FLAGS & GPIO_PULL_UP ? NRF_GPIO_PIN_PULLUP : \
+			      RECOVERY_BUTTON_FLAGS & GPIO_PULL_DOWN ? NRF_GPIO_PIN_PULLDOWN : \
+			      NRF_GPIO_PIN_NOPULL)
```

Comment:
this seems very odd, the user should set the flags and the flags should be applied as they are, not filtered, then the gpio functions should be used to determine if the button is pressed (which will take the GPIO_ACTIVE_* flag into consideration)

Reply:
If we are not using the zephyr GPIO driver, and the flag are passed directly to `nrf_gpio_reconfigure`, they must be converted.